### PR TITLE
Add Draw & Report

### PR DESCRIPTION
**Pull request by kdeloach**
* Add draw & report plugin to layer selector
* Upgrade old dojo API to use latest AMD API
* Report on all active layers
* Fix tooltips

Connects #588

### CODE REVIEW

**Comment by kdeloach** [comment]
There's a few design changes I'm planning to make on Friday but I believe this is feature complete.

**Comment by caseypt** [comment]
Taking a look at this now.

File: src/GeositeFramework/plugins/layer_selector_v2/draw_report/main.js [hunk 1]
```diff
@@ -0,0 +1,356 @@
...
+            getActiveLayers: function() {
+                return this.getTree().getSelectedLeafNodes();
+            },
+
+            // Called when layers are toggled.
+			update: function() {
+                this.setReportData(null);
+                this.queueRequestReport();
+                this.render();
+			},
```
Comment (caseypt) [inline src/GeositeFramework/plugins/layer_selector_v2/draw_report/main.js]: Indentation is off for this function.

**Comment by caseypt** [comment]
If you add layers that have reports, draw an area of interest, then turn off the layers, the area of interest remains on the map. This is not a big deal, but consider hiding it.

**Comment by caseypt** [comment]
+1. This works really well. I was able to draw many AoIs, generate reports, and download results without error. Nice job. This was a lot of work.

File: src/GeositeFramework/plugins/layer_selector_v2/draw_report/main.js (hunk 1 above)
Comment (kdeloach) [inline src/GeositeFramework/plugins/layer_selector_v2/draw_report/main.js]: Thanks, my editor has been misbehaving lately.

**Comment by kdeloach** [comment]
Let me ask Jeff what he thinks about deleting or preserving the area of interest after clicking the "Reset Layers" button.

**Comment by mmcfarland** [comment]
+1, this looks great.

I think we may need a clear button for just the AoI, too, but I think that can be included on future touch-up cards.  This looks feature complete to me.

**Comment by kdeloach** [comment]
Thanks.

**Pull request merged**